### ISSUE TITLE
Which library includes the functions like PyArg_ParseTuple() ?

### ISSUE THREAD

**Issue by anthonyhu81** [ninia/jep#282]
Hi,
I got the following error while trying to import my module on AIX (32bit):
  *Exception:rtld: 0712-001 Symbol PyArg_ParseTuple was referenced from module /opt/lib/python3.7/lib- dynload/_posixsubprocess.so(), but a runtime definition of the symbol was not found.*

And also cannot find PyBytes_AsString(), PyErr_Fetch(), PyErr_Restore(), PyErr_SetFromErrno(), PyErr_SetString(), PyExc_OSError().
I have the libpython3.7m.a path included in $LIBPATH and also the path /opt/lib/python3.7, did I miss any other libraries?

And which library includes these functions?

Thanks!

**Comment by anthonyhu81** [comment]
Also have python 2.7 installed on this machine, the default python is python2.7, the python2.7 libraries locate in /usr/lib, I'm not sure if this is impacted by python2.7.

**Comment by bsteffensmeier** [comment]
Those symbols are defined in libpython. I don't know much about AIX library loading but I suspect you may need to set the LDR_PRELOAD to ensure that the symbols are loaded into the global space.

**Comment by anthonyhu81** [comment]
Thank you bsteffensmeier,
In AIX only have the library file named libpython3.7m.a, seems it is a static library. I do not find any shared library files for python.
Did JEP link this library?
I'm trying to build jep in my AIX machine, but got the following warning:
ld: 0711-327 WARNING: Entry point not found: PyInit_jep
ld: 0711-224 WARNING: Duplicate symbol: .PyEval_SaveThread
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_Initialize
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_CallTracing
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_EvalCodeWithName
ld: 0711-224 WARNING: Duplicate symbol: .PyEval_EvalCodeEx
ld: 0711-224 WARNING: Duplicate symbol: .PyEval_EvalCode
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_SignalAsyncExc
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_SliceIndexNotNone
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_SliceIndex
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_RequestCodeExtraIndex
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_GetSwitchInterval
ld: 0711-224 WARNING: Duplicate symbol: ._PyEval_SetSwitchInterval
ld: 0711-224 WARNING: Duplicate symbol: .PyEval_ReInitThreads
ld: 0711-224 WARNING: Duplicate symbol: .PyEval_ReleaseThread
.............

Any idea?

**Comment by bsteffensmeier** [comment]
I do not recommend trying to build jep with a static python library. If you built python from source you can build a shared library by passing "--enable-shared" to the configure command.

**Comment by anthonyhu81** [comment]
I installed python3.7 on my AIX machine, and then install jep by command "pip3 install jep".
but when I run my application, I got the errors that cannot find these symbols such as "PyArg_ParseTuple".

I have python2.7.10 installed in this machine, and the python2.7 library and bin files locate in /usr/lib and /usr/bin, python3.7 files locate in /opt/lib and /opt/bin, is it possible that, the jep linked python2.7 library while installing？
 So I want to build the jep myself to check the linked library file.
I think it used the correct library, following is the output of jep building
  -ljvm -lpython3.7m -ldl -o build/lib.aix-7.1-3.7/jep/jep.so -L/opt/lib
but still have the warning  like:
  ld: 0711-327 WARNING: Entry point not found: PyInit_jep
  ld: 0711-224 WARNING: Duplicate symbol: .PyEval_SaveThread
